duckduckgo/Android
# Implement automatic fireproofing experiment

Task/Issue URL: https://app.asana.com/0/72649045549333/1200337281476877/f

### Description
This adds an on-boarding CTA to enable or disable automatic fireproofing.

### Steps to test this PR

**Code changes before testing**: Change the weighting of the "mi" variant in the `VariantManager` to 0.0

🟢 _Enabled Flow_
1. Freshly install the app.
2. Go to _example.com_.
3. Tap the fire button and clear all data.
4. Verify that the fireproofing CTA is shown (Pixel: `m_fireproof_cta_shown`).
5. Tap "Keep Me Signed In To Sites" (Pixel: `m_fireproof_cta_keep_me_signed_in_button`).
6. Verify that the final onboarding CTA is shown.
7. Go to _gmail.com_ and log in.
8. Verify that a Snackbar is shown indicating that the site has been fireproofed (Pixel: `m_fireproof_snackbar_shown`).
9. Verify that Gmail has been added to the list in "Settings" -> "Fireproof Sites" and that the toggle is "on".

🔴 _Disabled flow_
1. Freshly install the app.
2. Go to _example.com_.
3. Tap the fire button and clear all data.
4. Verify that the fireproofing CTA is shown (Pixel: `m_fireproof_cta_shown`).
5. Tap "Burn Everything" (Pixel: `m_fireproof_cta_burn_everything_button`).
6. Verify that the final onboarding CTA is shown.
7. Go to _gmail.com_ and log in.
8. The site should log in without fireproofing.
9. Verify that Gmail has not been added to "Settings" -> "Fireproof Sites" and that the toggle is "off".

👉 _Manual toggling_
1. Tap the fire button and clear all data.
2. Go to "Settings" -> "Fireproof Sites" and turn the "Fireproof Sites When Signing In" toggle to "on".
2. Go to _gmail.com_ and log in.
3. Verify that a Snackbar is shown indicating that the site has been fireproofed (Pixel: `m_fireproof_snackbar_shown`).
4. Verify that Gmail has been added to the list in "Settings" -> "Fireproof Sites".

### UI changes

**New CTA**

<img width="426" alt="fire_proof_cta" src="https://user-images.githubusercontent.com/3471025/139436611-58efa946-c9e2-4f95-bc5e-ef275b5a59c1.png">

| Setting Before  | Setting After |
| ------ | ----- |
<img width="426" alt="before_fireproof_settings" src="https://user-images.githubusercontent.com/3471025/139436368-15172f46-30fd-46c4-a341-388817b004e4.png">|<img width="425" alt="after_fireproof_settings" src="https://user-images.githubusercontent.com/3471025/139436468-c1f68a02-6006-468d-8052-616b7c7bcbc0.png">
